### ISSUE TITLE
How to find the corresponding values for "kUSBSleepPortCurrentLimit" and "kUSBWakePortCurrentLimit" in SSDT-EC-USBX

### ISSUE THREAD

**Issue by vuxuanhungg** [dreamwhite/bugtracker#41]
Hi, it seems interesting to me that your SSDT-EC-USBX discards 2 desktop properties in the USBX. However, comparing your two values of "kUSBSleepPortCurrentLimit" and "kUSBWakePortCurrentLimit" to the original code by `acidanthera`, they are different.

Your version:
```dsl
Return (Package (0x04)
{
    "kUSBSleepPortCurrentLimit", 
    0x0BB8, 
    "kUSBWakePortCurrentLimit", 
    0x0BB8
})
```

`acidanthera`'s version:
```dsl
Return (Package (0x04)
{
    "kUSBSleepPortCurrentLimit",
    0x0834,
    "kUSBWakePortCurrentLimit",
    0x0834
})
```

I wonder how to get the appropriate values for those two keys. There's not much info on this topic. [The only one I found](https://www.tonymacx86.com/threads/guide-usb-power-property-injection-for-sierra-and-later.222266/post-1563737) seems to suggest I need an ioreg or an ACPI dump from the real mac model with the same SMBIOS version.

**Comment by dreamwhite** [comment]
Good morning and thank you so much for creating this issue. Actually, as far as I'm aware, the values proposed in my `SSDT-EC-USBX` under `USBX` device were calculated by the following:

- the two discarded values (`kUSBSleepPowerSupply` and `kUSBWakePowerSupply`) were dropped for obvious reasons: it's a laptop, not a desktop
- the other values instead, if you convert them from hex to dec return `3000` which is obviously an higher value rather than acidanthera's version. I don't know if that's the ampere or other things but as discussed many times in the forums, theoretically speaking you should adapt those values in a way that connecting an iPad won't lead to a `USB Current Limit` warning in macOS.

Hope I clarified your doubts

**Comment by vuxuanhungg** [comment]
Could you give me the link to the post discussing the two discarded values? I'm surprised that `Dortania` does not distinguish the SSDT between laptop and desktop or even mention this matter.

Regarding the two other values, are those only meaningful if I connect an iPad to the laptop (or is it irrelevant if I only use an iPhone or Android phone)?

**Comment by dreamwhite** [comment]
> Could you give me the link to the post discussing the two discarded values? I'm surprised that `Dortania` does not distinguish the SSDT between laptop and desktop or even mention this matter.

I'm still looking for it. I think I have saved it somewhere. Will let you know


> Regarding the two other values, are those only meaningful if I connect an iPad to the laptop (or is it irrelevant if I only use an iPhone or Android phone)?

As far as I'm aware it's meaningful if you connect an iPad. Still looking for documentation :/

## UPDATE

I found this: https://github.com/acidanthera/bugtracker/issues/1001#issuecomment-650047317

Apparently you can test other devices also which have greater power consumption

**Comment by vuxuanhungg** [comment]
I tried loading two versions to see the differences. In IOReg, both cases only show `kUSBSleepPortCurrentLimit` and `kUSBWakePortCurrentLimit`.
![Screen Shot 2023-11-29 at 11 54 38](https://github.com/dreamwhite/bugtracker/assets/89293664/02d8296d-95d4-4f66-92eb-ba34de82a8af)

It also seems like on MacbookPro, only the two properties are present. The MacbookPro14,3 in [this example](https://www.tonymacx86.com/threads/guide-usb-power-property-injection-for-sierra-and-later.222266/) proves that (the same as the post you referred to).

Regarding the value, I started with 3000 and it's been working stable for days. Today, I switched to 2100 and it also seems to work fine. Actually, I don't see any different in `About This Mac` > `System Report` > `USB`. For example:
- USB 3.0: `Current available` is 500mA, `Current required` is 296mA.
- USB 2.0: `Current available` is 500mA, `Current required` is 200mA.
- For Android phone: `Current available` is 900mA, `Current required` is 500mA.

It seems like with different devices, macOS will adjust `Current available` accordingly. (I don't own an iPhone so cannot test that tho)
I even tried with 800mA (0x0320), and the above results stay the same. However, I encountered sleep issue due to the improper USB power. So I will stick with either 2100 or 3000 for now.

Edit: add image
